### PR TITLE
Feature/roe 144 add web validation logic

### PR DESCRIPTION
**Pull request by mouhajer-ch**
### JIRA link

Resolves [ROE-378](https://companieshouse.atlassian.net/browse/ROE-378), [ROE-305](https://companieshouse.atlassian.net/browse/ROE-305), [ROE-303](https://companieshouse.atlassian.net/browse/ROE-303) and [ROE-246](https://companieshouse.atlassian.net/browse/ROE-246)

### Change description

 - Add validation logic through the middleware
 - Integrate validation for the Presenter page
 - Integrate validation for the Entity page
 - Integrate validation for the BO statements page
 - Integrate validation for the BO type page
 - Update GETs controller and Views with flat address object

### Work checklist

- [x] Tests added where applicable
- [x] UI changes meet accessibility criteria

### CODE REVIEW

File: src/controllers/beneficial.owner.gov.controller.ts [hunk 1]
```diff
@@ -1,21 +1,28 @@
...
 import { StartDateKey, StartDateKeys } from "../model/date.model";
+import { BeneficialOwnerGovKey, BeneficialOwnerGovKeys } from "../model/beneficial.owner.gov.model";
 
 export const get = (req: Request, res: Response, next: NextFunction) => {
   try {
     logger.debugRequest(req, `GET ${config.BENEFICIAL_OWNER_GOV_PAGE}`);
 
     const appData: ApplicationData = getApplicationData(req.session);
 
+    const boGov = appData[BeneficialOwnerGovKey];
```
Comment (markpit) [inline src/controllers/beneficial.owner.gov.controller.ts:17]: boGov will be a list of boGov objects, so will this only display the fist one? how do we know we get the correct one from teh list?

Comment (mouhajer-ch) [reply src/controllers/beneficial.owner.gov.controller.ts:17]: Hi Mark, good point, until we do not have the update and delete logic we do not visualise(GET) previous data for BOs and MOs

File: src/middleware/validation.middleware.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import { NAVIGATION } from "../utils/navigation";
+
+export function checkValidations(req: Request, res: Response, next: NextFunction) {
+  const errorList = validationResult(req);
+
+  if (!errorList.isEmpty()) {
+    const errors = formatValidationError(errorList.array());
+
+    return res.render(NAVIGATION[req.path].currentPage, {
+      backLinkUrl: NAVIGATION[req.path].previusPage,
```
Comment (markpit) [inline src/middleware/validation.middleware.ts:14]: typo in previusPage, should be previousPage

File: src/controllers/beneficial.owner.gov.controller.ts (hunk 1 above)
Comment (mouhajer-ch) [reply src/controllers/beneficial.owner.gov.controller.ts:17]: I created this POC [here](https://github.com/companieshouse/overseas-entities-web/compare/feature/roe-147-simplify-adding-types-of-beneficial-owners) for updating and deleting BOs and MOs

Comment (mouhajer-ch) [reply src/controllers/beneficial.owner.gov.controller.ts:17]: But as you have suggested probably better remove the logic for the flat address fields right now

**Review by markpit — APPROVED** [review]

**Pull request merged**